tests/thread_safe_reference.cpp

```diff
@@ -668,18 +668,15 @@ TEST_CASE("thread safe reference") {
             REQUIRE(results.is_valid());
             REQUIRE(results.size() == 0);
         }
-#if 0
+
         SECTION("int results") {
             r->begin_transaction();
             obj = create_object(r, "int array", {{"value", AnyVector{INT64_C(1)}}});
-            List list(r, *get_table(*r, "int array"), 0, 0);
+            List list(r, obj.obj(), get_table(*r, "int array")->get_column_key("value"));
             r->commit_transaction();
 
-            auto results = delete_and_resolve(list.sort({{"self", true}}));
-            REQUIRE(results.is_valid());
-            REQUIRE(results.size() == 0);
+            REQUIRE(!delete_and_resolve(list).is_valid());
```

Comment:
This behavior is inconsistent with lists of objects, but consistent with the non-handover case so I think the list-of-objects one is the one that's wrong.

Reply:
I agree, this looks like correct behaviour here to me.
Isn't the difference with the list-of-objects test above because it is testing the `Results` returned by sorting the list and not the list itself? We probably haven't even implemented getting a `Results` on a list of primitives yet (other than a list of objects).